messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistSdk.kt

```diff
@@ -1,339 +1,122 @@
-
-    internal var gistQueue: Queue = Queue()
-    internal var gistModalManager: GistModalManager = GistModalManager()
-    internal var currentRoute: String = ""
-
-    // Global CoroutineScope for Gist Engine, all Gist classes should use same scope
-    internal var coroutineScope: CoroutineScope = GlobalScope
-
-    private val inAppMessagingManager = SDKComponent.inAppMessagingManager
-
-    // Global WebViewClientInterceptor for Gist Engine, used to intercept WebViewClient events while testing
-    internal var engineWebViewClientInterceptor: EngineWebViewClientInterceptor? = null
-
-    @JvmStatic
-    fun getInstance() = this
+interface GistProvider {
+    fun setCurrentRoute(route: String)
+    fun setUserId(userId: String)
+    fun dismissMessage()
+    fun reset()
+}
 
-    internal fun onActivityResumed(activity: Activity) {
-        resumedActivities.add(activity.javaClass.name)
+class GistSdk(
+    private val application: Application,
+    siteId: String,
+    dataCenter: String,
+    environment: GistEnvironment = GistEnvironment.PROD
+) : GistProvider {
+    private val inAppMessagingManager: InAppMessagingManager
+        get() = SDKComponent.inAppMessagingManager
+    private val state: InAppMessagingState
+        get() = inAppMessagingManager.getCurrentState()
+    private val globalPreferenceStore: GlobalPreferenceStore
+        get() = SDKComponent.android().globalPreferenceStore
+    private val logger = SDKComponent.logger
 
-        // Start polling if app is resumed and user messages are not being observed
-        val isNotObservingMessages =
-            observeUserMessagesJob == null || observeUserMessagesJob?.isCancelled == true
+    private var timer: Timer? = null
+    private val gistQueue = SDKComponent.gistQueue
 
-        if (isAppResumed() && getUserToken() != null && isNotObservingMessages) {
-            observeMessagesForUser()
-        }
+    private fun resetTimer() {
+        timer?.cancel()
+        timer = null
     }
 
-    internal fun onActivityPaused(activity: Activity) {
-        resumedActivities.remove(activity.javaClass.name)
-
-        // Stop polling if app is in background
-        if (!isAppResumed()) {
-            observeUserMessagesJob?.cancel()
-            observeUserMessagesJob = null
-        }
+    private fun onActivityResumed() {
+        logger.debug("Activity resumed, starting polling")
+        fetchInAppMessages(state.pollInterval)
     }
 
-    fun init(
-        application: Application,
-        siteId: String,
-        dataCenter: String,
-        environment: GistEnvironment = GistEnvironment.PROD
-    ) {
-        GistSdk.application = application
-        GistSdk.siteId = siteId
-        GistSdk.dataCenter = dataCenter
-        isInitialized = true
-        gistEnvironment = environment
-
-        subscribeToLifecycleEvents()
+    private fun onActivityPaused() {
+        logger.debug("Activity paused, stopping polling")
+        resetTimer()
```

Comment:
I think this can break things. Previously, we stopped timer only when activities list was empty, this seems to stop on pause state for every activity? Also, if user never stays for 10 seconds, will this never fetch as on every screen change, the timer is stopped 🤔

Reply:
nope onResume, makes the network call first and then run the timer.